scripts/generate-method-action-types.ts

```diff
@@ -0,0 +1,536 @@
+      return null;
+    }
+
+    return {
+      name: context.className,
+      filePath,
+      exposedMethods: context.exposedMethods,
+      methods: context.methods,
+    };
+  } catch (error) {
+    console.error(`Error parsing ${filePath}:`, error);
+    return null;
+  }
+}
+
+/**
+ * Extracts JSDoc comment from a method declaration.
+ *
+ * @param node - The method declaration node.
+ * @param sourceFile - The source file.
+ * @returns The JSDoc comment.
+ */
+function extractJSDoc(
+  node: ts.MethodDeclaration,
+  sourceFile: ts.SourceFile,
+): string {
+  const jsDocTags = ts.getJSDocCommentsAndTags(node);
+  if (jsDocTags.length === 0) {
+    return '';
+  }
+
+  const jsDoc = jsDocTags[0];
+  if (ts.isJSDoc(jsDoc)) {
+    const fullText = sourceFile.getFullText();
+    const start = jsDoc.getFullStart();
+    const end = jsDoc.getEnd();
+    const rawJsDoc = fullText.substring(start, end).trim();
+    return formatJSDoc(rawJsDoc);
+  }
+
+  return '';
+}
+
+/**
+ * Formats JSDoc comments to have consistent indentation for the generated file.
+ *
+ * @param rawJsDoc - The raw JSDoc comment from the source.
+ * @returns The formatted JSDoc comment.
+ */
+function formatJSDoc(rawJsDoc: string): string {
+  const lines = rawJsDoc.split('\n');
+  const formattedLines: string[] = [];
+
+  for (let i = 0; i < lines.length; i++) {
+    const line = lines[i];
+    if (i === 0) {
+      // First line should be /**
+      formattedLines.push('/**');
+    } else if (i === lines.length - 1) {
+      // Last line should be */
+      formattedLines.push(' */');
+    } else {
+      // Middle lines should start with ' * '
+      const trimmed = line.trim();
+      if (trimmed.startsWith('*')) {
+        // Remove existing * and normalize
+        const content = trimmed.substring(1).trim();
+        formattedLines.push(content ? ` * ${content}` : ' *');
+      } else {
+        // Handle lines that don't start with *
+        formattedLines.push(trimmed ? ` * ${trimmed}` : ' *');
+      }
+    }
+  }
+
+  return formattedLines.join('\n');
+}
+
+/**
+ * Extracts method signature as a string for the handler type.
+ *
+ * @param node - The method declaration node.
+ * @returns The method signature.
+ */
+function extractMethodSignature(node: ts.MethodDeclaration): string {
+  // Since we're just using the method reference in the handler type,
+  // we don't need the full signature - just return the method name
+  // The actual signature will be inferred from the controller class
+  return node.name ? (node.name as ts.Identifier).text : '';
+}
+
+/**
+ * Generates the action types file for a controller.
+ *
+ * @param controller - The controller information object.
+ */
+async function generateActionTypesFile(
+  controller: ControllerInfo,
+): Promise<void> {
+  const outputDir = path.dirname(controller.filePath);
+  const outputFile = path.join(
+    outputDir,
+    `${controller.name}-method-action-types.ts`,
```

Comment:
Just found something: when adding `MESSENGER_EXPOSED_METHODS` to SampleGasPricesController, it seems that we name this file `SampleGasPricesController-method-action-types.ts` when the file name for the controller in that package is called `sample-gas-prices-controller.ts`.

Maybe we want this to be based on the original file name rather than the controller name?